packages/data-stores/src/stepper-internal/reducer.ts

```diff
@@ -1,13 +1,23 @@
 import { combineReducers } from '@wordpress/data';
+import { SiteIntent } from '../onboard/constants';
 import { StepperInternalAction } from './actions';
 import type { Reducer } from 'redux';
 
-export const stepData: Reducer< any, StepperInternalAction > = ( state = null, action ) => {
+type StepData = Record< string, unknown > & {
+	path: string;
+	intent: SiteIntent;
+	previousStep: string;
+};
+
+export const stepData: Reducer< StepData | undefined, StepperInternalAction > = (
+	state = undefined,
+	action
+) => {
 	if ( action.type === 'SET_STEP_DATA' ) {
 		return action.data;
 	}
 	if ( action.type === 'CLEAR_STEP_DATA' ) {
-		return null;
+		return undefined;
```

Comment:
Why is it needed to change from null to undefined?

Reply:
I forgot the reason so I reverted the changes.